### PR TITLE
libgpod: reduce propagated libraries, cleanup

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libgpod/default.nix [hunk 1]
```diff
@@ -27,20 +40,30 @@ stdenv.mkDerivation rec {
...
+  propagatedBuildInputs = [
+    libimobiledevice
+  ];
+
+  meta = with lib; {
+    homepage = "https://sourceforge.net/projects/gtkpod/";
     description = "Library used by gtkpod to access the contents of an ipod";
-    license = "LGPL";
-    platforms = lib.platforms.gnu ++ lib.platforms.linux;
+    license = licenses.lgpl2;
```
Comment (jtojnar) [inline pkgs/development/libraries/libgpod/default.nix:65]: It appears to be `lgpl21Plus`: https://sourceforge.net/p/gtkpod/libgpod/ci/master/tree/src/itdb_sqlite.c

File: pkgs/development/libraries/libgpod/default.nix [hunk 2]
```diff
@@ -27,20 +40,30 @@ stdenv.mkDerivation rec {
...
 
-  dontStrip = true;
-
-  propagatedBuildInputs = [ glib libxml2 sqlite zlib sg3_utils
-    gdk-pixbuf taglib libimobiledevice ];
+  dontStrip = monoSupport;
 
   nativeBuildInputs = [ autoreconfHook intltool pkg-config ]
     ++ (with perlPackages; [ perl XMLParser ])
     ++ lib.optionals monoSupport [ mono gtk-sharp-2_0 ];
```
Comment (jtojnar) [inline pkgs/development/libraries/libgpod/default.nix:47]: `gtk-sharp-2_0` should probably go to `buildInputs`.

File: pkgs/development/libraries/libgpod/default.nix [hunk 3]
```diff
@@ -27,20 +40,30 @@ stdenv.mkDerivation rec {
     "--with-udev-dir=${placeholder "out"}/lib/udev"
   ] ++ lib.optionals monoSupport [ "--with-mono" ];
 
-  dontStrip = true;
-
-  propagatedBuildInputs = [ glib libxml2 sqlite zlib sg3_utils
```
Comment (jtojnar) [inline pkgs/development/libraries/libgpod/default.nix:32]: `lib/pkgconfig/libgpod-1.0.pc` contains `Requires: glib-2.0 >= 2.8.0 gobject-2.0 gdk-pixbuf-2.0 libimobiledevice-1.0`

File: pkgs/development/libraries/libgda/default.nix [hunk 4]
```diff
@@ -65,6 +61,10 @@ stdenv.mkDerivation rec {
     postgresql
   ];
 
+  propagatedBuildInputs = [
+    libxml2
```
Comment (jtojnar) [inline pkgs/development/libraries/libgda/default.nix:65]: `lib/pkgconfig/libgda-5.0.pc`: `Requires: gobject-2.0 gthread-2.0 libxml-2.0`

File: pkgs/development/libraries/libgpod/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/libgpod/default.nix:32]: There is no gobject right now. Not sure if this should be added.

File: pkgs/development/libraries/libgda/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/libgda/default.nix:65]: gthread is glib? and should we move gobject from native to propagated?

File: pkgs/development/libraries/libgpod/default.nix (hunk 3 above)
Comment (jtojnar) [reply pkgs/development/libraries/libgpod/default.nix:32]: Yeah, gobject and gthread are all part of glib.

**Pull request merged**